tests/e2e_test/status_propagation_test.go

```diff
@@ -53,12 +54,44 @@ var _ = Describe("Warning Status Propagation", Ordered, func() {
 			Should(Succeed())
 	})
 
-	It("Should disable Template Operator and Kyma should result in Ready status", func() {
+	It("Should propagate Warning state even after resource is marked for deletion", func() {
+		By("Getting manifest object key")
+		manifestObjectKey, err := GetManifestObjectKey(ctx, controlPlaneClient,
+			kyma.GetName(), "template-operator")
+		Expect(err).To(Not(HaveOccurred()))
+		By("Getting resource object key")
+		resourceObjectKey, err := GetResourceObjectKey(ctx, controlPlaneClient, manifestObjectKey)
+		Expect(err).To(Not(HaveOccurred()))
+		By("Adding finalizer to avoid deletion")
+		Eventually(AddFinalizerToSampleResource).
+			WithContext(ctx).
+			WithArguments(resourceObjectKey, controlPlaneClient, "e2e-finalizer").
+			Should(Succeed())
+
 		By("Disabling Template Operator")
 		Eventually(DisableModule).
 			WithContext(ctx).
 			WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", runtimeClient).
 			Should(Succeed())
+
+		By("Checking manifest state")
+		Eventually(CheckManifestIsInState).
+			WithContext(ctx).
+			WithArguments(manifestObjectKey, controlPlaneClient, declarative.StateWarning).
+			Should(Succeed())
+		By("Checking module state in Kyma CR")
+		Eventually(CheckKymaModuleIsInState).
+			WithContext(ctx).
+			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient,
+				"template-operator", v1beta2.StateWarning).
+			Should(Succeed())
+
+		By("Remove finalizer")
+		err = RemoveFinalizerToSampleResource(ctx, resourceObjectKey, controlPlaneClient, "e2e-finalizer")
+		Expect(err).To(Not(HaveOccurred()))
+	})
+
+	It("Kyma should result in Ready status", func() {
```

Comment:
after remove finalizer, kyma should result in Ready status. They should put in the same `It` spec, but not separate one.